src/java/picard/sam/markduplicates/util/OpticalDuplicateFinder.java

```diff
@@ -96,13 +99,14 @@ public int compare(final PhysicalLocation lhs, final PhysicalLocation rhs) {
             for (int j = i + 1; j < length; ++j) {
                 final PhysicalLocation rhs = list.get(j);
 
-                if (opticalDuplicateFlags[j]) continue;
+                if ((rhs == keeper && opticalDuplicateFlags[i]) || opticalDuplicateFlags[j]) continue;
```

Comment:
`(rhs == keeper && opticalDuplicateFlags[i])` can actually be moved into the outer loop.